com.unity.render-pipelines.high-definition/Editor/AssetProcessors/MaterialPostProcessor.cs

```diff
@@ -16,16 +16,56 @@ class MaterialModificationProcessor : AssetModificationProcessor
     {
         static void OnWillCreateAsset(string asset)
         {
-            if (!asset.ToLowerInvariant().EndsWith(".mat"))
+            if (asset.ToLowerInvariant().EndsWith(".mat"))
+            {
+                MaterialPostprocessor.s_CreatedAssets.Add(asset);
                 return;
+            }
 
-            MaterialPostprocessor.s_CreatedAssets.Add(asset);
+            // For .shadergraph assets, this is tricky since the callback will be for the .meta
+            // file only as we don't create it with AssetDatabase.CreateAsset but later AddObjectToAsset
+            // to the .shadergraph via the importer context.
+            // At the time the meta file is created, the .shadergraph is already present
+            // but Load*AtPAth(), GetMainAssetTypeAtPath() etc. won't find anything.
+            // The GUID is already present though, and we actually use those facts to infer we
+            // have a newly created shadergraph.
+
+            // TODO TOCHECK:
+            // HDMetaData subasset will be included after SG creation anyway so unlike for materials
+            // (cf .mat with AssetVersion in OnPostprocessAllAssets) we dont need to manually add a subasset.
+            // For adding them to MaterialPostprocessor.s_ImportedAssetThatNeedSaving for SaveAssetsToDisk()
+            // to make them editable (flag for checkout), we still detect those here, but not sure this is
+            // helpful as right now, even on re-import, a .shadergraph multijson is not rewritten, so only
+            // /Library side serialized data is actually changed (including the generated .shader that can
+            // also change which is why we run shadergraph reimports), and re-import from the same .shadergraph
+            // should be idempotent.
+            if (asset.ToLowerInvariant().EndsWith($".{ShaderGraphImporter.Extension}.meta"))
```

Comment:
Essentially this is to decide if shadegraph assets need to resave or not when updated, so "MakeEditable" can be called on the asset. But since we currently dont even rewrite the .shadergraph json text itself when reimporting (only library side effects happen: the subtarget does update and all, the embedded shader and HDMetaData objects change but we don't rewrite the .shadergraph on disk), maybe it can be removed.

Reply:
For your awareness: https://github.com/Unity-Technologies/Graphics/pull/4508, not sure if it change anything for your PR

Reply:
I don't think it does, but I can't be sure